client/utils/useShowDelegationModalOption.tsx

```diff
@@ -0,0 +1,23 @@
+/* OgvStaking contract requires for each of the accounts to perform delegation to their
+ * own address in order for propose and cast vote function to work properly on the contract.
+ *
+ */
+import { useStore } from "utils/store";
+import { ZERO_ADDRESS } from "constants/index";
+
+const useShowDelegationModalOption = () => {
+  const { ogvDelegateeAddress } = useStore();
+
+  // returns bool -> true if delegation modal needs to be shown
+  return async () => {
+    // account already has delegation set
+    const needToShowDelegation = ogvDelegateeAddress === ZERO_ADDRESS;
+    if (needToShowDelegation) {
+      useStore.setState({ ensureDelegationModalOpened: true });
+    }
+
+    return needToShowDelegation;
```

Comment:
This does not return `false` after registering as a delegate within the modal, then closing the modal and click Submit Proposal. 

It only acts as intended post browser refresh.

We probably need a method to reload the state on transaction confirmation here.